src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,709 @@
+﻿using System;
+using System.Reactive.Disposables;
+using System.Windows.Input;
+using Avalonia.Controls.Metadata;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.LogicalTree;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button with primary and secondary parts that can each be pressed separately.
+    /// The primary part behaves like a <see cref="Button"/> and the secondary part opens a flyout.
+    /// </summary>
+    [PseudoClasses(
+        ":disabled",
+        ":secondaryButtonRight",
+        ":secondaryButtonSpan",
+        ":checkedFlyoutOpen",
+        ":flyoutOpen",
+        ":checkedTouchPressed",
+        ":checked",
+        ":checkedPrimaryPressed",
+        ":checkedPrimaryPointerOver",
+        ":checkedSecondaryPressed",
+        ":checkedSecondaryPointerOver",
+        ":touchPressed",
+        ":primaryPressed",
+        ":primaryPointerOver",
+        ":secondaryPressed",
+        ":secondaryPointerOver")]
+    public class SplitButton : ContentControl, ICommandSource
```

Comment:
I think it's worth to inherit it from Button to make it consistent with any other *Button control. At least it would be expected, and I don't see any blockers for that.

Reply:
WinUI specifically decided NOT to inherit from Button. I'm not sure exactly why but I am sure they have a good reason. I really don't want to change the inheritance for this unless we know for sure it's a good idea.

Reply:
It's simple enough control though.
We can keep it for now for further discussion, if anybody else has more opinions.

Reply:
I'm wondering if the default pointer logic from Button would then interfere with the SplitButton since it has two buttons within it's template - so you'd have to change the pointer logic and essentially end up with a new control anyway? Also not sure the reasoning

Reply:
With routed events when internal button handles pointer input, parent split button won't receive it.
But it should be tested also with other input methods and access keys.

Reply:
Comment made here: https://github.com/AvaloniaUI/Avalonia/pull/7422#issuecomment-1021911616

 > I still think it worth to reuse code from Button class. On other hand we might need to consider creating ButtonBase instead and move some common code there. But it should be discussed internally first and most likely done in separated PR.

Reply:
If you want to do that in the future, ok. Do make it a separate PR though. I would also strongly encourage you and the internal team to attempt to understand why Microsoft did what they did. They don't do these types of things without reasons and something like control inheritance is heavily vetted internally.